GameModel.md

```diff
@@ -0,0 +1,78 @@
+
+
+Stage 3
+---
+_(not strictly in this order, can be inserted earlier)_
+
+Potential Verifiers make Initial Deposits (not necessa
+
+rily equal to v), declaring that they agree with the contract terms. The actual Verifier is chosen randomy among them and signs the contract for Work - this ends stage 3. Note thah the gathering and /or selection of Verifier could have been done at any time earlier; the timing is not critical, but it affects the length of time that the Verifier's stake is locked. There is a minor practical trade-off: choosing a Verifier late rather than early allows to minimize the time that the Verifier's stake is locked, however it creates more uncertainty whether the Verifier will be found in time.
+
+If the chosen Verifier fails to sign the contract by a certain time, then he is penalized with the size of his Initial Deposit. This penalty prevents the stalling attack. The money from penalty could be burnt or given to the Client and/or Worker as a compensation for wasting time. 
+
+
+Stage 4.
+---
+
+The Verifier receives the information how to calculate the Sample. The Verifier calculates the Sample and reports to the Client. Hence, the Client gets the information whether the Worker's results are considered correct by the Verifier - this ends stage 4. Note that it is not critical whether the Verifier received the Sample value calculated by the Worker. Both settings are possible: a) where the Verifier reports his own Sample result (without knowing the Worker's Sample result) and b) where the Verifier reports only a binary agreement or disagreement with the Worker's Sample result. However, the setting (a) is better, as it makes collusion more costly.
+
+If the Verifier fails to report to the Client by a certain time, then he is penalized (not necessarily with a full stake v, however it should be a significant penalty, not a minor one). This penalty prevents the stalling attack and other combined attacks. In this case, a replacement random Verifier needs to take over the contract, perhaps one that is chosen initially as a replacement. The money from the penalty could be burnt or given to the Client and/or Worker  as a compensation for wasting time. 
```

Comment:
If the whole stake of Verifier will be split between the Worker and the Client, wouldn't it create a wrong incentive to them to misreport the Verifier failure?

Reply:
Correct, and this is why stake money taken from W or V should never not be fully split between C, W, V - a large part of the stake should be burnt, while minor compensations can be given. The stakes exist mostly to provide payment for Arbitration, if such a need arises. Minor compensations to C, W, V may useful to counter-effect attacks that waste time of network participants (I call these "stalling attacks"). 

As for the quoted passage: I thought that the Verifier's failure to report to the Client can be detected automatically. If it needs to be confirmed by the Client, then it is better not to give to Client any monetary rewards for penalising the Verifier. And again, probably the Verifier should be able to appeal to Arbitrage.